connections/tests/utils.py

```diff
@@ -12,29 +14,93 @@ def make_apartments_sold_in_elastic() -> None:
     sleep(3)
 
 
-def get_elastic_apartments_for_sale_uuids() -> list:
+def get_elastic_apartments_for_sale_uuids(
+    etuovi=False, oikotie=False, only=False, return_UUID=False
+) -> list:
     s_obj = (
         Search()
         .query("match", _language="fi")
         .query("match", apartment_state_of_sale=ApartmentStateOfSale.FOR_SALE)
     )
+
+    if oikotie:
+        s_obj = s_obj.query("match", publish_on_oikotie=oikotie)
+        if only:
+            s_obj = s_obj.query("match", publish_on_etuovi=False)
+    if etuovi:
+        s_obj = s_obj.query("match", publish_on_etuovi=etuovi)
+        if only:
+            s_obj = s_obj.query("match", publish_on_oikotie=False)
+
+    if only and not etuovi and not oikotie:
+        s_obj = s_obj.query("match", publish_on_oikotie=oikotie).query(
+            "match", publish_on_etuovi=etuovi
+        )
+
     s_obj.execute()
     scan = s_obj.scan()
     uuids = []
     for hit in scan:
-        uuids.append(str(hit.uuid))
+        if return_UUID:
+            uuids.append(UUID(hit.uuid))
+        else:
+            uuids.append(hit.uuid)
```

Comment:
This function already has quite many things to do so, so I think it might be better if it returns only a single type of values (maybe `UUID` for better type-safety). If the caller needs strings, they can just do `str(value)` before comparisons.

Reply:
The downside of removing flag "return_UUID" is that there are few places where lists would be needed to convert to string. From my point of view, flag was more efficient solution for helper function, and not that extreme.

Reply:
I agree with @jmp. The result should only return a single type of value. If necessary, the method which calls this can make anything with the result value.